### PR TITLE
test(storage): add retry conformance tests [WIP]

### PR DESCRIPTION
**Pull request by tritone**

### CODE REVIEW

File: storage/conformance_test.go [hunk 1]
```diff
@@ -29,8 +34,191 @@ import (
...
+// 2. A list of non-idempotent calls (should give expectSuccess = False for
+//    everything).
+var idempotentOps = []retryFunc{
+	// storage.objects.get
+	func(ctx context.Context, c *Client) error {
+		_, err := c.Bucket(bucketName).Object(objName).Attrs(ctx)
+		return err
+	},
+	// GET object (XML API)
+	func(ctx context.Context, c *Client) error {
```
Comment (frankyn) [inline storage/conformance_test.go:69]: How do you reset the retry counter for this subsequent call?

File: storage/conformance_test.go [hunk 2]
```diff
@@ -29,8 +34,191 @@ import (
...
+		r, err := c.Bucket(bucketName).Object(objName).NewReader(ctx)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(ioutil.Discard, r)
+		return err
+	},
+	// storage.objects.update (preconditions)
```
Comment (frankyn) [inline storage/conformance_test.go:77]: Ah i see, you're including conditionally idempotent operations here as well

File: storage/conformance_test.go [hunk 3]
```diff
@@ -16,21 +16,315 @@ package storage
...
+	//}
+	return resp, err
+}
+
+// Create custom client that sends instruction
+func wrappedClient(host, testID string) (*Client, error) {
+	ctx := context.Background()
+	base := http.DefaultTransport
+	trans, err := htransport.NewTransport(ctx, base, option.WithScopes(raw.DevstorageFullControlScope),
+		option.WithUserAgent("custom-user-agent"))
```
Comment (frankyn) [inline storage/conformance_test.go:313]: You don't  need to supply `option`'s do you?

File: storage/conformance_test.go [hunk 4]
```diff
@@ -16,21 +16,315 @@ package storage
...
+		return nil, fmt.Errorf("failed to create http client: %v", err)
+	}
+	c := http.Client{Transport: trans}
+
+	// Add RoundTripper to the created HTTP client.
+	wrappedTrans := &withTestID{rt: c.Transport, testID: testID}
+	c.Transport = wrappedTrans
+
+	// Supply this client to storage.NewClient
+	client, err := NewClient(ctx, option.WithHTTPClient(&c), option.WithEndpoint(host + "/storage/v1/"))
```
Comment (frankyn) [inline storage/conformance_test.go:324]: Does with `WithEndpoint` support host only or does it need `/storage/v1/`*?

**Review by frankyn — CHANGES_REQUESTED** [review]

File: storage/conformance_test.go (hunk 3 above)
Comment (tritone) [reply storage/conformance_test.go:313]: Yeah that's true, it's only needed for the real service I believe. Will remove.

File: storage/conformance_test.go (hunk 4 above)
Comment (tritone) [reply storage/conformance_test.go:324]: It needs the full endpoint unfortunately.

**Pull request closed without merging**